sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2248,21 +2244,37 @@ public <T> Mono<FeedResponse<T>> readMany(
                         // partitionkeyrange
                         Map<PartitionKeyRange, SqlQuerySpec> rangeQueryMap;
                         rangeQueryMap = getRangeQueryMap(partitionRangeItemKeyMap,
+                            singleItemPartitionRequestMap,
                             collection.getPartitionKey());
 
+                        // create point reads
+                        Mono<? extends List<FeedResponse<Document>>> pointReads = singleItemPartitionRequestMap.values().size() > 0 ? createPointReadOperations(
```

Comment:
I feel these types can be simplified like this -> 

```
// create point reads
                        Flux<FeedResponse<Document>> pointReads = singleItemPartitionRequestMap.values().size() > 0 ? createPointReadOperations(
                            singleItemPartitionRequestMap,
                            resourceLink,
                            options,
                            klass) : Flux.empty();


                        // create the executable query
                        Flux<FeedResponse<Document>> queries = rangeQueryMap.keySet().size() > 0 ? createReadManyQuery(
                            resourceLink,
                            new SqlQuerySpec(DUMMY_SQL_QUERY),
                            options,
                            Document.class,
                            ResourceType.Document,
                            collection,
                            Collections.unmodifiableMap(rangeQueryMap)) : Flux.empty();
```

And then the other code review suggestions can be applied on top of this structure.